code/game/objects/items/devices/radio/intercom.dm

```diff
@@ -108,6 +108,8 @@
 	return ..()
 
 /obj/item/device/radio/intercom/attack_ai(mob/user as mob)
+	if(!ai_can_interact(user))
+		return
```

Comment:
why isn't this just on the parent of attack_ai with `SHOULD_CALL_PARENT(TRUE)` ?

Reply:
because the machinery level attack_ai calls attack_hand, not TRUE or FALSE, so it needs to be overridden